2025-01-14 20:35:02 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 22776 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:35:02 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:35:03 WARN  [main] o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
2025-01-14 20:35:03 ERROR [main] o.s.boot.SpringApplication - Application run failed
org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:185)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:418)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:290)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:349)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:118)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:791)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:609)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
	at ceu.dam.ad.ejerciciosTema3.avanzado.App.main(App.java:15)
Caused by: org.springframework.context.annotation.ConflictingBeanDefinitionException: Annotation-specified bean name 'seriesServiceImpl' for bean class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.SeriesServiceImpl] conflicts with existing, non-compatible bean definition of same name and class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.Correccion.SeriesServiceImpl]
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.checkCandidate(ClassPathBeanDefinitionScanner.java:361)
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.doScan(ClassPathBeanDefinitionScanner.java:288)
	at org.springframework.context.annotation.ComponentScanAnnotationParser.parse(ComponentScanAnnotationParser.java:128)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:332)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:267)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:193)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:171)
	... 12 common frames omitted
2025-01-14 20:36:40 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 19904 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:36:40 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:36:40 WARN  [main] o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
2025-01-14 20:36:40 ERROR [main] o.s.boot.SpringApplication - Application run failed
org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:185)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:418)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:290)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:349)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:118)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:791)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:609)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
	at ceu.dam.ad.ejerciciosTema3.avanzado.App.main(App.java:13)
Caused by: org.springframework.context.annotation.ConflictingBeanDefinitionException: Annotation-specified bean name 'seriesServiceImpl' for bean class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.SeriesServiceImpl] conflicts with existing, non-compatible bean definition of same name and class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.Correccion.SeriesServiceImpl]
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.checkCandidate(ClassPathBeanDefinitionScanner.java:361)
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.doScan(ClassPathBeanDefinitionScanner.java:288)
	at org.springframework.context.annotation.ComponentScanAnnotationParser.parse(ComponentScanAnnotationParser.java:128)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:332)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:267)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:193)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:171)
	... 12 common frames omitted
2025-01-14 20:37:51 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 17944 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:37:51 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:37:51 WARN  [main] o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
2025-01-14 20:37:51 ERROR [main] o.s.boot.SpringApplication - Application run failed
org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:185)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:418)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:290)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:349)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:118)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:791)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:609)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
	at ceu.dam.ad.ejerciciosTema3.avanzado.App.main(App.java:13)
Caused by: org.springframework.context.annotation.ConflictingBeanDefinitionException: Annotation-specified bean name 'seriesServiceImpl' for bean class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.SeriesServiceImpl] conflicts with existing, non-compatible bean definition of same name and class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.Correccion.SeriesServiceImpl]
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.checkCandidate(ClassPathBeanDefinitionScanner.java:361)
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.doScan(ClassPathBeanDefinitionScanner.java:288)
	at org.springframework.context.annotation.ComponentScanAnnotationParser.parse(ComponentScanAnnotationParser.java:128)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:332)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:267)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:193)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:171)
	... 12 common frames omitted
2025-01-14 20:38:25 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 13208 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:38:25 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:38:25 WARN  [main] o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
2025-01-14 20:38:25 ERROR [main] o.s.boot.SpringApplication - Application run failed
org.springframework.beans.factory.BeanDefinitionStoreException: Failed to parse configuration class [ceu.dam.ad.ejerciciosTema3.avanzado.App]
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:185)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:418)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:290)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:349)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:118)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:791)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:609)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
	at ceu.dam.ad.ejerciciosTema3.avanzado.App.main(App.java:13)
Caused by: org.springframework.context.annotation.ConflictingBeanDefinitionException: Annotation-specified bean name 'seriesServiceImpl' for bean class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.SeriesServiceImpl] conflicts with existing, non-compatible bean definition of same name and class [ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.service.Correccion.SeriesServiceImpl]
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.checkCandidate(ClassPathBeanDefinitionScanner.java:361)
	at org.springframework.context.annotation.ClassPathBeanDefinitionScanner.doScan(ClassPathBeanDefinitionScanner.java:288)
	at org.springframework.context.annotation.ComponentScanAnnotationParser.parse(ComponentScanAnnotationParser.java:128)
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:332)
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:267)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:193)
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:171)
	... 12 common frames omitted
2025-01-14 20:39:56 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 22816 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:39:56 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:39:57 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2025-01-14 20:39:57 WARN  [main] o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.support.BeanDefinitionOverrideException: Invalid bean definition with name 'serieRepository' defined in ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.repository.SerieRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Cannot register bean definition [Root bean: class=org.springframework.data.jpa.repository.support.JpaRepositoryFactoryBean; scope=; abstract=false; lazyInit=false; autowireMode=0; dependencyCheck=0; autowireCandidate=true; primary=false; fallback=false; factoryBeanName=null; factoryMethodName=null; initMethodNames=null; destroyMethodNames=null; defined in ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.repository.SerieRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration] for bean 'serieRepository' since there is already [Root bean: class=org.springframework.data.jpa.repository.support.JpaRepositoryFactoryBean; scope=; abstract=false; lazyInit=false; autowireMode=0; dependencyCheck=0; autowireCandidate=true; primary=false; fallback=false; factoryBeanName=null; factoryMethodName=null; initMethodNames=null; destroyMethodNames=null; defined in ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.repository.Correccion.SerieRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration] bound.
2025-01-14 20:39:57 INFO  [main] o.s.b.a.l.ConditionEvaluationReportLogger - 

Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2025-01-14 20:39:57 ERROR [main] o.s.b.d.LoggingFailureAnalysisReporter - 

***************************
APPLICATION FAILED TO START
***************************

Description:

The bean 'serieRepository', defined in ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.repository.SerieRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration, could not be registered. A bean with that name has already been defined in ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio1.repository.Correccion.SerieRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration and overriding is disabled.

Action:

Consider renaming one of the beans or enabling overriding by setting spring.main.allow-bean-definition-overriding=true

2025-01-14 20:40:32 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 23256 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:40:32 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:40:32 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2025-01-14 20:40:32 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 46 ms. Found 8 JPA repository interfaces.
2025-01-14 20:40:32 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2025-01-14 20:40:32 INFO  [main] com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.mariadb.jdbc.Connection@474821de
2025-01-14 20:40:32 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2025-01-14 20:40:32 INFO  [main] o.h.jpa.internal.util.LogHelper - HHH000204: Processing PersistenceUnitInfo [name: default]
2025-01-14 20:40:32 INFO  [main] org.hibernate.Version - HHH000412: Hibernate ORM core version 6.6.2.Final
2025-01-14 20:40:32 INFO  [main] o.h.c.i.RegionFactoryInitiator - HHH000026: Second-level cache disabled
2025-01-14 20:40:33 INFO  [main] o.s.o.j.p.SpringPersistenceUnitInfo - No LoadTimeWeaver setup: ignoring JPA class transformer
2025-01-14 20:40:33 INFO  [main] o.hibernate.orm.connections.pooling - HHH10001005: Database info:
	Database JDBC URL [Connecting through datasource 'HikariDataSource (HikariPool-1)']
	Database driver: undefined/unknown
	Database version: 11.5.2
	Autocommit mode: undefined/unknown
	Isolation level: undefined/unknown
	Minimum pool size: undefined/unknown
	Maximum pool size: undefined/unknown
2025-01-14 20:40:33 ERROR [main] o.s.o.j.LocalContainerEntityManagerFactoryBean - Failed to initialize JPA EntityManagerFactory: Property 'ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio3.modelo.Tienda.marca' is not a collection and may not be a '@OneToMany', '@ManyToMany', or '@ElementCollection'
2025-01-14 20:40:33 WARN  [main] o.s.c.a.AnnotationConfigApplicationContext - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Property 'ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio3.modelo.Tienda.marca' is not a collection and may not be a '@OneToMany', '@ManyToMany', or '@ElementCollection'
2025-01-14 20:40:33 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2025-01-14 20:40:33 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
2025-01-14 20:40:33 INFO  [main] o.s.b.a.l.ConditionEvaluationReportLogger - 

Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2025-01-14 20:40:33 ERROR [main] o.s.boot.SpringApplication - Application run failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Property 'ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio3.modelo.Tienda.marca' is not a collection and may not be a '@OneToMany', '@ManyToMany', or '@ElementCollection'
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1802)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:336)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:288)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:334)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:204)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:970)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
	at ceu.dam.ad.ejerciciosTema3.avanzado.App.main(App.java:13)
Caused by: org.hibernate.AnnotationException: Property 'ceu.dam.ad.ejerciciosTema3.avanzado.ejercicio3.modelo.Tienda.marca' is not a collection and may not be a '@OneToMany', '@ManyToMany', or '@ElementCollection'
	at org.hibernate.boot.model.internal.CollectionBinder.determineSemanticJavaType(CollectionBinder.java:1121)
	at org.hibernate.boot.model.internal.CollectionBinder.determineCollectionClassification(CollectionBinder.java:1013)
	at org.hibernate.boot.model.internal.CollectionBinder.createBinderAutomatically(CollectionBinder.java:855)
	at org.hibernate.boot.model.internal.CollectionBinder.getCollectionBinder(CollectionBinder.java:848)
	at org.hibernate.boot.model.internal.CollectionBinder.bindCollection(CollectionBinder.java:284)
	at org.hibernate.boot.model.internal.PropertyBinder.bindProperty(PropertyBinder.java:897)
	at org.hibernate.boot.model.internal.PropertyBinder.buildProperty(PropertyBinder.java:811)
	at org.hibernate.boot.model.internal.PropertyBinder.processElementAnnotations(PropertyBinder.java:732)
	at org.hibernate.boot.model.internal.EntityBinder.processIdPropertiesIfNotAlready(EntityBinder.java:1086)
	at org.hibernate.boot.model.internal.EntityBinder.handleIdentifier(EntityBinder.java:418)
	at org.hibernate.boot.model.internal.EntityBinder.bindEntityClass(EntityBinder.java:250)
	at org.hibernate.boot.model.internal.AnnotationBinder.bindClass(AnnotationBinder.java:401)
	at org.hibernate.boot.model.source.internal.annotations.AnnotationMetadataSourceProcessorImpl.processEntityHierarchies(AnnotationMetadataSourceProcessorImpl.java:257)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess$1.processEntityHierarchies(MetadataBuildingProcess.java:281)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:324)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:1431)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1502)
	at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:66)
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:390)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:419)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:400)
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.afterPropertiesSet(LocalContainerEntityManagerFactoryBean.java:366)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1849)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1798)
	... 14 common frames omitted
2025-01-14 20:41:17 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 17748 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:41:17 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:41:17 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2025-01-14 20:41:18 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 45 ms. Found 8 JPA repository interfaces.
2025-01-14 20:41:18 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2025-01-14 20:41:18 INFO  [main] com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.mariadb.jdbc.Connection@4f94e148
2025-01-14 20:41:18 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2025-01-14 20:41:18 INFO  [main] o.h.jpa.internal.util.LogHelper - HHH000204: Processing PersistenceUnitInfo [name: default]
2025-01-14 20:41:18 INFO  [main] org.hibernate.Version - HHH000412: Hibernate ORM core version 6.6.2.Final
2025-01-14 20:41:18 INFO  [main] o.h.c.i.RegionFactoryInitiator - HHH000026: Second-level cache disabled
2025-01-14 20:41:18 INFO  [main] o.s.o.j.p.SpringPersistenceUnitInfo - No LoadTimeWeaver setup: ignoring JPA class transformer
2025-01-14 20:41:18 INFO  [main] o.hibernate.orm.connections.pooling - HHH10001005: Database info:
	Database JDBC URL [Connecting through datasource 'HikariDataSource (HikariPool-1)']
	Database driver: undefined/unknown
	Database version: 11.5.2
	Autocommit mode: undefined/unknown
	Isolation level: undefined/unknown
	Minimum pool size: undefined/unknown
	Maximum pool size: undefined/unknown
2025-01-14 20:41:19 INFO  [main] o.h.e.t.j.p.i.JtaPlatformInitiator - HHH000489: No JTA platform available (set 'hibernate.transaction.jta.platform' to enable JTA platform integration)
2025-01-14 20:41:19 INFO  [main] o.s.o.j.LocalContainerEntityManagerFactoryBean - Initialized JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:41:19 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Started App in 1.941 seconds (process running for 2.423)
2025-01-14 20:42:09 WARN  [main] o.m.jdbc.message.server.ErrorPacket - Error: 4091-42S02: Unknown SEQUENCE: 'tiendas_seq'
2025-01-14 20:42:09 WARN  [main] o.h.e.jdbc.spi.SqlExceptionHelper - SQL Error: 4091, SQLState: 42S02
2025-01-14 20:42:09 ERROR [main] o.h.e.jdbc.spi.SqlExceptionHelper - (conn=1414) Unknown SEQUENCE: 'tiendas_seq'
2025-01-14 20:42:09 INFO  [SpringApplicationShutdownHook] o.s.o.j.LocalContainerEntityManagerFactoryBean - Closing JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:42:09 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2025-01-14 20:42:09 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
2025-01-14 20:47:22 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 15916 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:47:22 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:47:22 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2025-01-14 20:47:22 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 44 ms. Found 8 JPA repository interfaces.
2025-01-14 20:47:22 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2025-01-14 20:47:22 INFO  [main] com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.mariadb.jdbc.Connection@7d0d91a1
2025-01-14 20:47:22 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2025-01-14 20:47:22 INFO  [main] o.h.jpa.internal.util.LogHelper - HHH000204: Processing PersistenceUnitInfo [name: default]
2025-01-14 20:47:22 INFO  [main] org.hibernate.Version - HHH000412: Hibernate ORM core version 6.6.2.Final
2025-01-14 20:47:22 INFO  [main] o.h.c.i.RegionFactoryInitiator - HHH000026: Second-level cache disabled
2025-01-14 20:47:23 INFO  [main] o.s.o.j.p.SpringPersistenceUnitInfo - No LoadTimeWeaver setup: ignoring JPA class transformer
2025-01-14 20:47:23 INFO  [main] o.hibernate.orm.connections.pooling - HHH10001005: Database info:
	Database JDBC URL [Connecting through datasource 'HikariDataSource (HikariPool-1)']
	Database driver: undefined/unknown
	Database version: 11.5.2
	Autocommit mode: undefined/unknown
	Isolation level: undefined/unknown
	Minimum pool size: undefined/unknown
	Maximum pool size: undefined/unknown
2025-01-14 20:47:23 INFO  [main] o.h.e.t.j.p.i.JtaPlatformInitiator - HHH000489: No JTA platform available (set 'hibernate.transaction.jta.platform' to enable JTA platform integration)
2025-01-14 20:47:23 INFO  [main] o.s.o.j.LocalContainerEntityManagerFactoryBean - Initialized JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:47:24 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Started App in 1.99 seconds (process running for 2.501)
2025-01-14 20:47:28 WARN  [main] o.m.jdbc.message.server.ErrorPacket - Error: 1364-HY000: Field 'uuid_centro' doesn't have a default value
2025-01-14 20:47:28 WARN  [main] o.h.e.jdbc.spi.SqlExceptionHelper - SQL Error: 1364, SQLState: HY000
2025-01-14 20:47:28 ERROR [main] o.h.e.jdbc.spi.SqlExceptionHelper - (conn=1424) Field 'uuid_centro' doesn't have a default value
2025-01-14 20:47:28 INFO  [SpringApplicationShutdownHook] o.s.o.j.LocalContainerEntityManagerFactoryBean - Closing JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:47:28 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2025-01-14 20:47:28 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
2025-01-14 20:48:33 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 23988 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:48:33 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:48:34 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2025-01-14 20:48:34 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 47 ms. Found 8 JPA repository interfaces.
2025-01-14 20:48:34 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2025-01-14 20:48:34 INFO  [main] com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.mariadb.jdbc.Connection@7d0d91a1
2025-01-14 20:48:34 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2025-01-14 20:48:34 INFO  [main] o.h.jpa.internal.util.LogHelper - HHH000204: Processing PersistenceUnitInfo [name: default]
2025-01-14 20:48:34 INFO  [main] org.hibernate.Version - HHH000412: Hibernate ORM core version 6.6.2.Final
2025-01-14 20:48:34 INFO  [main] o.h.c.i.RegionFactoryInitiator - HHH000026: Second-level cache disabled
2025-01-14 20:48:34 INFO  [main] o.s.o.j.p.SpringPersistenceUnitInfo - No LoadTimeWeaver setup: ignoring JPA class transformer
2025-01-14 20:48:34 INFO  [main] o.hibernate.orm.connections.pooling - HHH10001005: Database info:
	Database JDBC URL [Connecting through datasource 'HikariDataSource (HikariPool-1)']
	Database driver: undefined/unknown
	Database version: 11.5.2
	Autocommit mode: undefined/unknown
	Isolation level: undefined/unknown
	Minimum pool size: undefined/unknown
	Maximum pool size: undefined/unknown
2025-01-14 20:48:35 INFO  [main] o.h.e.t.j.p.i.JtaPlatformInitiator - HHH000489: No JTA platform available (set 'hibernate.transaction.jta.platform' to enable JTA platform integration)
2025-01-14 20:48:35 INFO  [main] o.s.o.j.LocalContainerEntityManagerFactoryBean - Initialized JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:48:35 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Started App in 1.865 seconds (process running for 2.352)
2025-01-14 20:49:20 INFO  [SpringApplicationShutdownHook] o.s.o.j.LocalContainerEntityManagerFactoryBean - Closing JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:49:20 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2025-01-14 20:49:20 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
2025-01-14 20:51:31 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Starting App using Java 23 with PID 5652 (C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados\target\classes started by adominguez5871 in C:\Users\adominguez5871\Desktop\JavaRepository\2ºTrimestre\Acceso a datos\ejercicios-tema3-avanzados)
2025-01-14 20:51:31 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - No active profile set, falling back to 1 default profile: "default"
2025-01-14 20:51:31 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2025-01-14 20:51:31 INFO  [main] o.s.d.r.c.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 45 ms. Found 8 JPA repository interfaces.
2025-01-14 20:51:32 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2025-01-14 20:51:32 INFO  [main] com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.mariadb.jdbc.Connection@4c86da0c
2025-01-14 20:51:32 INFO  [main] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2025-01-14 20:51:32 INFO  [main] o.h.jpa.internal.util.LogHelper - HHH000204: Processing PersistenceUnitInfo [name: default]
2025-01-14 20:51:32 INFO  [main] org.hibernate.Version - HHH000412: Hibernate ORM core version 6.6.2.Final
2025-01-14 20:51:32 INFO  [main] o.h.c.i.RegionFactoryInitiator - HHH000026: Second-level cache disabled
2025-01-14 20:51:32 INFO  [main] o.s.o.j.p.SpringPersistenceUnitInfo - No LoadTimeWeaver setup: ignoring JPA class transformer
2025-01-14 20:51:32 INFO  [main] o.hibernate.orm.connections.pooling - HHH10001005: Database info:
	Database JDBC URL [Connecting through datasource 'HikariDataSource (HikariPool-1)']
	Database driver: undefined/unknown
	Database version: 11.5.2
	Autocommit mode: undefined/unknown
	Isolation level: undefined/unknown
	Minimum pool size: undefined/unknown
	Maximum pool size: undefined/unknown
2025-01-14 20:51:32 INFO  [main] o.h.e.t.j.p.i.JtaPlatformInitiator - HHH000489: No JTA platform available (set 'hibernate.transaction.jta.platform' to enable JTA platform integration)
2025-01-14 20:51:32 INFO  [main] o.s.o.j.LocalContainerEntityManagerFactoryBean - Initialized JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:51:33 INFO  [main] c.d.ad.ejerciciosTema3.avanzado.App - Started App in 1.984 seconds (process running for 2.565)
2025-01-14 20:52:19 INFO  [SpringApplicationShutdownHook] o.s.o.j.LocalContainerEntityManagerFactoryBean - Closing JPA EntityManagerFactory for persistence unit 'default'
2025-01-14 20:52:19 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2025-01-14 20:52:19 INFO  [SpringApplicationShutdownHook] com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
